src/_pytest/python_api.py

```diff
@@ -954,15 +965,43 @@ def raises(
             f"Raising exceptions is already understood as failing the test, so you don't need "
             f"any special code to say 'this should never raise an exception'."
         )
+
+    expected_exceptions: tuple[type[E], ...]
+    origin_exc: type[E] | None = get_origin(expected_exception)
     if isinstance(expected_exception, type):
-        expected_exceptions: tuple[type[E], ...] = (expected_exception,)
+        expected_exceptions = (expected_exception,)
+    elif origin_exc and issubclass(origin_exc, BaseExceptionGroup):
+        expected_exceptions = (cast(type[E], expected_exception),)
     else:
         expected_exceptions = expected_exception
-    for exc in expected_exceptions:
-        if not isinstance(exc, type) or not issubclass(exc, BaseException):
+
+    def validate_exc(exc: type[E]) -> type[E]:
+        origin_exc: type[E] | None = get_origin(exc)
+        if origin_exc and issubclass(origin_exc, BaseExceptionGroup):
+            exc_type = get_args(exc)[0]
+            if issubclass(origin_exc, ExceptionGroup) and exc_type is Exception:
+                return cast(type[E], origin_exc)
+            elif (
+                issubclass(origin_exc, BaseExceptionGroup) and exc_type is BaseException
```

Comment:
I'd be inclined to do `... exc_type in (Any, (Base)Exception):` here - the escape-hatch is particularly useful when you might want to do something specific with the contents later.

Reply:
Sure, fixed in patch (left as own commit I assume you'll squash)